search/search_repository.go

```diff
@@ -685,6 +693,22 @@ func (r *GormSearchRepository) listItemsFromDB(ctx context.Context, criteria cri
 		return nil, 0, errors.NewBadParameterError("expression", criteria)
 	}
 
+	if childIterations {
+		itrRepo := iteration.NewIterationRepository(r.db)
+
+		where += fmt.Sprintf(` AND `)
+		allitrs := []string{}
+		for _, itr := range iterations {
```

Comment:
@baijum Do you think it is possible to have repeated iterations in this array?
If we have an array like `[itr1, itr2, itr3, itr1]` from the filter expression, we might want to change the code following this for loop. Maybe just a simple check that we don't have duplicates in the `allitrs` array should be fine. 

In a normal case, this `itr` repetition wouldn't happen, but if the query is has repetitions we should deal with it.

Reply:
Even if duplicate happens, it should be fine as we are doing `OR`.